client/components/site-icon/index.jsx

```diff
@@ -70,6 +75,19 @@ const SiteIcon = React.createClass( {
 	}
 } );
 
-export default connect( ( state, { siteId } ) => (
-	siteId ? { site: getSite( state, siteId ) } : {}
-) )( SiteIcon );
+export default connect( ( state, { site, siteId, imgSize } ) => {
+	// Until sites state is completely within Redux, we provide compatibility
+	// in cases where site object is passed to use the icon.img property as URL
+	if ( site ) {
+		return {
+			iconUrl: get( site, 'icon.img' )
+		};
+	}
+
+	// Otherwise, assume we want to perform the lookup in Redux state
+	// exclusively using the site ID
+	return {
+		site: getSite( state, siteId ),
+		iconUrl: getSiteIconUrl( state, siteId, imgSize )
```

Comment:
This selector relies on two sub-state trees: sites and media, Shouldn't we add Queries to the component? probably a `QuerySites` since "in theory" the component could get any siteId as prop. I and guess `QueryMedia` is probably not available yet and receiveMedia is only called if we use the `SiteIconSetting` component, so I wonder how is it filled in that case?